scripts/datastore_upgrade.py

```diff
@@ -0,0 +1,311 @@
+import json
+import os
+import subprocess
+import sys
+import time
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../lib'))
+import appscale_info
+from constants import CONTROLLER_SERVICE
+from constants import APPSCALE_HOME
+import monit_interface
+
+sys.path.append(os.path.join(os.path.dirname(__file__), '../AppDB'))
+import appscale_datastore_batch
+import datastore_server
+from dbconstants import *
+from zkappscale import zktransaction as zk
+
+sys.path.append(os.path.join(os.path.dirname(__file__), "../AppServer"))
+from google.appengine.datastore import entity_pb
+
+# The location on the local filesystem where Cassandra is installed.
+CASSANDRA_DIR = "/opt/cassandra"
+
+# The location of the Cassandra binary on the local filesystem.
+CASSANDRA_EXECUTABLE = CASSANDRA_DIR + "/cassandra/bin/cassandra"
+
+PID_FILE = "/var/appscale/appscale-cassandra.pid"
+
+# The default port to connect to ZooKeeper.
+ZK_PORT = ":2181"
+
+# The location on the local filesystem where we should store ZooKeeper data.
+ZK_DATA_LOCATION = "/opt/appscale/zookeeper"
+
+# The maximum number of entities to be returned from the query.
+_MAX_ENTITIES = 1000000
+
+def ensure_app_is_not_running():
+    """ Ensures AppScale is not running as this is an offline script. """
+    print ("Ensure AppScale is not currently running...")
+    appscale_running = subprocess.call(['service', CONTROLLER_SERVICE, 'status']) == 0
+    if appscale_running:
+        print ("AppScale is running, please shut it down and try again.")
+        exit(1)
+
+def start_cassandra():
+    """ Creates a monit configuration file and prompts Monit to start Cassandra. """
+    print ("Starting Cassandra...")
```

Comment:
Since the `logging` module is more configurable, we typically use it instead of print statements.